packages/sdk/src/identity/EthereumKeyPairIdentity.ts

```diff
@@ -0,0 +1,65 @@
+import { SignerWithProvider } from './Identity'
+import { binaryToHex, ECDSA_SECP256K1_EVM, hexToBinary } from '@streamr/utils'
+import { Wallet } from 'ethers'
+import { RpcProviderSource } from '../RpcProviderSource'
+import { SignatureType } from '@streamr/trackerless-network'
+import { KeyPairIdentityConfig, StrictStreamrClientConfig } from '../Config'
+import { KeyPairIdentity } from './KeyPairIdentity'
+
+/**
+ * An Identity that derives an Ethereum address from a secp256k1 private key
+ * and uses that as the UserID instead of the actual public key.
+ */
+export class EthereumKeyPairIdentity extends KeyPairIdentity {
+    constructor(privateKey: string, address?: string) {
+        const impliedAddress = new Wallet(privateKey).address.toLowerCase()
+        super(
+            hexToBinary(impliedAddress), 
+            hexToBinary(privateKey)
+        )
+        if (address && address.toLowerCase() !== impliedAddress) {
+            throw new Error(`The given publicKey does not match the privateKey! The privateKey implies address: ${impliedAddress}`)
+        }
+    }
+
+    // eslint-disable-next-line class-methods-use-this
+    assertKeyPairIsValid(): void {
+        // Ensured by constructor
+    }
+
+    // eslint-disable-next-line class-methods-use-this
+    getSignatureType(): SignatureType {
+        return SignatureType.ECDSA_SECP256K1_EVM
+    }
+
+    // eslint-disable-next-line class-methods-use-this
+    getExpectedPublicKeyLength(): number {
+        // Address, not actual public key
```

Comment:
Is there some specific need that this returns the length of the address, not the length of the public key (i.e. 64)?